python/src/aiconfig/eval/common.py

```diff
@@ -11,64 +12,36 @@
 T_OutputDatum = TypeVar("T_OutputDatum", contravariant=True)
 
 
+@dataclass
+class CustomMetricValue(ABC):
+    """Subclass this if you want your metric to return a type not included in MetricValue."""
```

Comment:
Maybe a comment here as well on ensuring the comparator operators are implemented?